.gitlab-ci.yml

```diff
@@ -0,0 +1,95 @@
+variables:
+  PROJECT_ALLOC_NAME: ${CI_PROJECT_NAME}_ci_${CI_PIPELINE_ID}
+  BUILD_ROOT: ${CI_PROJECT_DIR}
+  # Should make it so we aren't doing fresh clones of GEOSX all the time
+  GIT_STRATEGY: fetch
+  GIT_SUBMODULE_STRATEGY: normal
+
+stages:
+  - build
+  - nightly
+
+####
+# Template
+.build_script:
+  script:
+    - mkdir ${SYSTEM}
+
+    # CONFIGURE
+    - echo "~~~~~~~~~~ START - configure ~~~~~~~~~~~"
+    - python scripts/config-build.py -hc host-configs/LLNL/${HOST_CONFIG} -bt Release -DENABLE_DOXYGEN=OFF 2>&1 | tee ${SYSTEM}/${SYSTEM}_configure.log || config_code=$?
+    - if [ $config_code -ne 0 ]; then mv ${SYSTEM}/${SYSTEM}_configure.log ${SYSTEM}/${SYSTEM}_configure_FAILURE.log; else mv ${SYSTEM}/${SYSTEM}_configure.log ${SYSTEM}/${SYSTEM}_configure_SUCCESS.log; fi
```

Comment:
@TotoGaz  Yes, ideally that was what I wanted as well and use the REST API... Unfortunately, the generated private tokens are only available for 24 hours then deleted the same day, and `$CI_JOB_TOKEN` is not available for LC gitlab repositories (non-public).

Reply:
> the generated private tokens are only available for 24 hours then deleted the same day

That's strange. Nothing that you can do to circumvent this restrictive configuration?

Reply:
Not currently, it's by design in gitlab:
https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#when-personal-access-tokens-expire

Reply:
https://docs.gitlab.com/ee/user/admin_area/settings/account_and_limit_settings.html#limit-the-lifetime-of-access-tokens 
This link seem to indicate that this could be configured?

Reply:
Feature isn't available for me. Doesn't seem available for LC Gitlab, or only for LC Admins.